### PR TITLE
Features/hdf 8 view stats per pool

### PR DESCRIPTION
**Pull request by schystz**
## Description of Changes

Display the following stats per pool on Pools page:
- total pool value ($)
- BPT balance
- BPT staked value ($)
- HALO rewards earned

Other changes/fixes:
- fixed a bug in `develop` branch when approving the stake amount

Under the hood changes:
- updated halo hooks to consolidate all useful info in `poolsInfo` state (from `useBalancer` hook), removing all redundant logic from other hooks (e.g. `usePoolSummary`), making the app more efficient
- created a currency formatter helper to easily display currency amounts on the UI

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-8)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/SOOW-XXX-description`
* [x] All files have appropriate file headers and documentation
* ~~[ ] I have written new tests for your core changes, as applicable~~
* ~~[ ] I have successfully ran tests locally~~

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] Code style confirms to eslint/flow rules
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -83,6 +82,23 @@ export const BALANCER_SUBGRAPH_URL =
...
+    'usd-coin': '0x2F375e94FC336Cdec2Dc0cCB5277FE59CBf1cAe5',
+    augur: '0x8c9e6c40d3402480ACE624730524fACC5482798c',
+    'wrapped-bitcoin': '0xe0C9275E44Ea80eF17579d33c55136b7DA269aEb',
+    'basic-attention-token': '0x1f1f156E0317167c11Aa412E3d1435ea29Dc3cCE',
+    havven: '0x86436BcE20258a6DcfE48C9512d4d49A30C4d8c4',
+    aragon: '0x37f03a12241E9FD3658ad6777d289c3fb8512Bc9',
+    '0x': '0xccb0F4Cf5D3F97f4a55bb5f5cA321C3ED033f244'
+  }
+}
+
```
Comment (tracyarciaga) [inline src/constants/index.ts:112]: Do we need all these tokens here? ^^

**Comment by tracyarciaga** [comment]
@schystz trying to test this feature branch, but the when I click the "Stake" on the main  nav bar, it just redirects to pool page.

**Comment by schystz** [comment]
> @schystz trying to test this feature branch, but the when I click the "Stake" on the main nav bar, it just redirects to pool page.

@tracyarciaga it was behind develop thats why the stake tab isn't working (we merged Ernest's PR just yesterday). I have rebased from develop so both tabs should be working now.

File: src/halo-hooks/usePoolsSummary.ts [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
   })
 
   // Get user balance for each pool token
-  const balances = useTokenBalances(account ?? undefined, poolTokens)
+  const poolsAsTokens = poolsInfo.map(poolInfo => poolInfo.asToken)
+  const balances = useTokenBalances(account ?? undefined, poolsAsTokens)
 
   // Get totalSupply of each pool token
-  const totalSupplies = useTokenTotalSuppliesWithLoadingIndicator(poolTokens)[0]
+  const totalSupplies = useTokenTotalSuppliesWithLoadingIndicator(poolsAsTokens)[0]
```
Comment (ernestpascual) [inline src/halo-hooks/usePoolsSummary.ts:16]: just a question why did you use poolsAsTokens?

**Review by ernestpascual — CHANGES_REQUESTED** [review]
Hi @schystz sorry took time to verify some stuff:


Can we fix the HALO value to 2 decimal places? it seems to be breaking the UI. You can mint your own HALO from the scripts but i can send some as well.
Screen Shot 2021-04-06 at 11 08 54 AM

Im getting this errors, do you get this as well? this is when staking and unstaking BPT
Screen Shot 2021-04-06 at 2 09 03 PM

**Comment by schystz** [comment]
Thanks for the catch @ernestpascual ! Yeah I agree, HALO earnings need to be formatted to 2 decimal places, like what we are doing with BPT tokens.

I didn't notice the staking error before when I run it locally. I was able to stake and unstake even. But let me check again now to be sure.

File: src/halo-hooks/usePoolsSummary.ts (hunk 2 above)
Comment (schystz) [reply src/halo-hooks/usePoolsSummary.ts:16]: Naming the variable `poolTokens` might be confused with `pool.tokens` (balancer pool tokens) so what I ended up doing instead is I created a new property `pool.asToken` then collectively called them `poolsAsTokens` (which represents an array of uniswap's `Token` object).

Comment (schystz) [reply src/halo-hooks/usePoolsSummary.ts:16]: Screen Shot 2021-04-06 at 3 51 53 PM

Sharing this for further reference on the poolInfo object structure

Comment (ernestpascual) [reply src/halo-hooks/usePoolsSummary.ts:16]: thanks for the reference! agreeing with this as well.

**Comment by schystz** [comment]
@ernestpascual updated with the following changes:

- refactored `usePoolsSummary` hook to move away from `useEffect` as it doesn't play nicely with `useBalances` and `useTokenTotalSuppliesWithLoadingIndicator` causing an endless loop. To achieve this, I used `useMemo` and created 3 new hooks to get: user total claimed HALO, user BPT staked per pool, user unclaimed HALO per pool (by using other uniswap hooks as pattern)

- added the 2 decimal place rounding when showing HALO balance

**Review by tracyarciaga — APPROVED** [review]

File: src/constants/index.ts (hunk 1 above)
Comment (schystz) [reply src/constants/index.ts:112]: @tracyarciaga to give a better context - these are kovan addresses of popular tokens so we can map them out from coingecko prices (coingecko only keep tracks of mainnet addresses), which is a step required in able to display total pool value in the pools table. So yeah, we do need them here :) Feel free to suggest though if you think we should define this on another place.

**Review by ernestpascual — APPROVED** [review]
given the gas thing bug will be resolved on another issue, ill approve this pr @schystz  @tracyarciaga

**Comment by schystz** [comment]
Thanks @tracyarciaga & @ernestpascual ! Merging this now to develop.

**Pull request merged**